examples/embeddings/computeEmbeddings.py

```diff
@@ -0,0 +1,266 @@
+# !pip install pydgraph pybars3 sentence_transformers mistralai openai
+import sys
+import json
+import os
+import re
+import pydgraph
+from openai import OpenAI
+from mistralai.client import MistralClient
+
+from pybars import Compiler
+from sentence_transformers import SentenceTransformer, util
+
+
+# Example of embeddings.json
+#{
+# "embeddings" : [
+#   {     
+#     "entityType":"Product",
+#     "attribute":"product_embedding",
+#     "index":"hnsw(metric: \"euclidean\")",
+#     "provider": "huggingface",
+#     "model":"sentence-transformers/all-MiniLM-L6-v2",
+#     "config" : {
+#         "dqlQuery" : "{ title:Product.title }",
+#         "template": "{{title}} "
+#     },
+#     "disabled": false
+#   }
+# ]
+# }
+#
+#  provider : huggingface or openai or mistral
+#  model : model name from the provider doing embeddings
+
+# TODO 
+# check code for template loops.
+
+compiler = Compiler()
+global client # dgrpah client is a global variable
+
+assert "DGRAPH_GRPC" in os.environ, "DGRAPH_GRPC must be defined"
+dgraph_grpc = os.environ["DGRAPH_GRPC"]
+if "cloud.dgraph" in dgraph_grpc:
+    assert "DGRAPH_ADMIN_KEY" in os.environ, "DGRAPH_ADMIN_KEY must be defined"
+    APIAdminKey = os.environ["DGRAPH_ADMIN_KEY"]
+else:
+    APIAdminKey = None
+
+# TRANSFORMER_API_KEY must be defined in env variables
+# client stub for on-prem requires grpc host:port without protocol
+# client stub for cloud requires the grpc endpoint of graphql endpoint or base url of the cluster
+# to run on a self-hosted env, unset ADMIN_KEY and set DGRAPH_GRPC
+
+def setClient():
+    global client
+    if APIAdminKey is None:
+      client_stub = pydgraph.DgraphClientStub(dgraph_grpc)
+    else:
+        client_stub = pydgraph.DgraphClientStub.from_cloud(dgraph_grpc,APIAdminKey )     
+    client = pydgraph.DgraphClient(client_stub)
+
+def clearIndex(predicate):
+    print(f"remove index for {predicate}")
+    schema = f"{predicate}: float32vector ."
+    op = pydgraph.Operation(schema=schema)
+    alter = client.alter(op)
+    print(alter)
+def computeIndex(predicate,index):
+    print(f"create index for {predicate} {index}")
+    schema = f"{predicate}: float32vector @index({index}) ."
+    op = pydgraph.Operation(schema=schema)
+    alter = client.alter(op)
+    print(alter) 
+
+def huggingfaceEmbeddings(model,sentences):
+    embeddings = model.encode(sentences)
+    return embeddings.tolist()
+
+
+def computeEmbedding(predicate,data, template,provider,modelName,model,llm, dimensions):
```

Comment:
Some formatting issues here and throughout the file. You should run a linter over this perhaps. I recommend [ruff](https://docs.astral.sh/ruff/)

Reply:
Thanks. I'm using it now !